app/src/main/java/io/homeassistant/companion/android/settings/SettingsFragment.kt

```diff
@@ -81,6 +89,21 @@ class SettingsFragment : PreferenceFragmentCompat(), SettingsView {
         SensorWorker.start(context!!)
     }
 
+    override fun onDisplayPreferenceDialog(preference: Preference) {
```

Comment:
Once the `SsidDialogFragment` works like other preferences this shouldn't be needed.

Reply:
Implemented the requested changes, for this one I changed it to the official supported way by implementing the `OnPreferenceDisplayDialogCallback` interface in the `SettingsActivity`.